modules/apps/portal-workflow/portal-workflow-kaleo-runtime-integration-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/integration/internal/WorkflowTaskManagerImpl.java

```diff
@@ -121,13 +120,24 @@ public WorkflowTask assignWorkflowTaskToUser(
 			long companyId, long userId, long workflowTaskId,
 			long assigneeUserId, String comment, Date dueDate,
 			Map<String, Serializable> workflowContext)
-		throws WorkflowException {
+		throws PortalException {
 
 		PermissionChecker permissionChecker =
 			PermissionThreadLocal.getPermissionChecker();
 
-		if (permissionChecker.getUserId() != userId) {
-			ReflectionUtil.throwException(new PrincipalException());
+		List<User> assignableUsers = ListUtil.toList(
+			getAssignableUsers(companyId, workflowTaskId));
+
+		assignableUsers.add(_userLocalService.getUser(userId));
```

Comment:
Hi @IstvanD, sorry for delay

This line looks a little confusing to me. Could you explain to me why you are manually adding the user passed as a parameter in the list of assignable users?

Thank you in advance

Reply:
Hi @seltong 
89
I'm sorry if the code may look confusing. The assigned user is added to the list because `getAssignableUsers()` won't contain this user. If `assignWorkflowTaskToUser()` is called from a workflow script without setting up a `permissionChecker`, the condition `permissionChecker.getUserId() != userId` will be `true` and the call will fail with a `PrincipalException`.

Reply:
Hi @IstvanD,

The condition `permissionChecker.getUserId() != userId` already exists,

to finish the fix just remove this `assignableUsers.add(_userLocalService.getUser(userId));`

Thank you,
Feliphe

Reply:
to avoid principal exception you can set the userId as you mentioned or you need to call the LocalServiceUtil methods step by step.